### ISSUE TITLE
VAAPI transcode failed

### ISSUE THREAD

**Issue by No-Zi** [jellyfin/jellyfin#3094]
**Describe bug**
I want to transcode media with an AMD GPU.
When i start a command manualy is work but when jellyfin transcode he failed.

**System:**
 - OS: [Debian]
 - Virtualization: [KVM,]
 - Clients: [Browser, Android]
 - Browser: [ Firefox 76]
 - Jellyfin Version: [10.5.5]
 - Playback: [Transcode] 
 - Installed Plugins: [AudioDB, Gotify Notifications, MusicBrainz,TMDb Box Sets]
 - Reverse Proxy: [nginx]
 - Base URL: []
 - Networking: [NAT]
 - Storage: [NFS]

**Logs**
[AVHWDeviceContext @ 0x5560146fb940] libva: /usr/lib/x86_64-linux-gnu/dri/radeonsi_drv_video.so has no function __vaDriverInit_1_0
[AVHWDeviceContext @ 0x5560146fb940] Failed to initialise VAAPI connection: -1 (unknown libva error).
Device creation failed: -5.
Failed to set value '/dev/dri/renderD128' for option 'vaapi_device': Input/output error
Error parsing global options: Input/output error

**Screenshots**
![image](https://user-images.githubusercontent.com/65168974/81561167-0913f580-9393-11ea-8318-957955cf6f56.png)

**Comment by Artiume** [comment]
```
[AVHWDeviceContext @ 0x5560146fb940] libva: /usr/lib/x86_64-linux-gnu/dri/radeonsi_drv_video.so has no function __vaDriverInit_1_0
[AVHWDeviceContext @ 0x5560146fb940] Failed to initialise VAAPI connection: -1 (unknown libva error).
Device creation failed: -5.
Failed to set value '/dev/dri/renderD128' for option 'vaapi_device': Input/output error
Error parsing global options: Input/output error
```

**Comment by joelcollet** [comment]
Hi 
don't know much about jellyfin, but had almost same issue with ffmpeg on AMD (failed init).
solved by (re)installing mesa libs . as well i added the user launching ffmpeg to the video group (or use sudo..)
Joël

**Comment by No-Zi** [comment]
>  Joël

Hi
Thanks for your reply, I've remove all packages of mesa and reinstall and it don't work...
The user jellyfin have the group video and render

**Comment by Optimizer-Prime** [comment]
I had a similar error in my logs when trying to setup my RX580 to do hardware acceleration with VAAPI and was getting the same error in your screenshot. I'm on Ubuntu 20.04 running Jellyfin from the repo.

What codec are you trying to transcode from? I've seen bug reports of HEVC transcoding not really working currently and this is a mesa issue, although it's fixed in 20.1. It seems I was able to get it working and the log files say that VAAPI is being used now to transcode, even 10bit HEVC files.

Here's a quick overview of what I did:
1.) installed the latest ffmpeg on Ubuntu
2.) under Server > Playback > FFmpeg path, set it to '/usr/bin/ffmpeg'. Or something along that line, you need to use the system install, not the default Jellyfin version.
3.) Add the Oibaf repo (or another one that would have the latest mesa drivers)
```
sudo add-apt-repository ppa:oibaf/graphics-drivers
sudo apt-get update
```
4.) run `glxinfo | grep "OpenGL version"` to see what version of mesa you currently have.
5.) run `sudo apt-get dist-upgrade` to install the mesa drivers. Best to restart your computer after this assuming it finishes without errors.
6.) Perform step 4 again to confirm that the mesa drivers have updated, they should be at least version 20.1
7.) try running a HEVC file in browser or something that would need to transcode and then check the log file to see if it's using VAAPI.

After doing this I was able to get it working. However, it seems to transcode at a slower FPS than software in most cases with HEVC. My CPU is an EPYC 7351p for reference. I'm also unable to transcode large 4k 10bit HEVC HDR files smoothly, or even unplayable because the transcoding FPS is <24, with or without hardware accel. I'm sure that'll improve with time. But most HEVC files transcode just fine with VAAPI, at a CRF of 20. 

Hopefully this will be of some help though!

**Comment by No-Zi** [comment]
Thanks Optimizer-Prime it's work ! 

For one film it not work, it's a 4K HEVC film and an other it freez. It's probably an incompability of my Graphic card

Error code :
Stream mapping:
  Stream #0:0 -> #0:0 (hevc (native) -> h264 (h264_vaapi))
  Stream #0:1 -> #0:1 (ac3 (native) -> aac (native))
Press [q] to stop, [?] for help
[h264_vaapi @ 0x55abeaa8b440] Driver does not support some wanted packed headers (wanted 0xd, found 0).
[h264_vaapi @ 0x55abeaa8b440] Driver does not support packed sequence headers, but a global header is requested.
[h264_vaapi @ 0x55abeaa8b440] No global header will be written: this may result in a stream which is not usable for some purposes (e.g. not muxable to some containers).
frame=   14 fps=0.0 q=0.0 size=N/A time=-577014:32:22.77 bitrate=N/A speed=N/A
frame=   24 fps= 19 q=0.0 size=N/A time=-577014:32:22.77 bitrate=N/A speed=N/A
frame=   34 fps= 18 q=0.0 size=N/A time=-577014:32:22.77 bitrate=N/A speed=N/A
frame=   45 fps= 17 q=0.0 size=N/A time=-577014:32:22.77 bitrate=N/A speed=N/A
frame=   56 fps= 17 q=0.0 size=N/A time=-577014:32:22.77 bitrate=N/A speed=N/A
frame=   69 fps= 18 q=0.0 size=N/A time=-577014:32:22.77 bitrate=N/A speed=N/A
frame=   81 fps= 18 q=0.0 size=N/A time=-577014:32:22.77 bitrate=N/A speed=N/A
frame=   98 fps= 19 q=0.0 size=N/A time=-577014:32:22.77 bitrate=N/A speed=N/A
frame=  111 fps= 19 q=0.0 size=N/A time=-577014:32:22.77 bitrate=N/A speed=N/A
frame=  125 fps= 19 q=0.0 size=N/A time=-577014:32:22.77 bitrate=N/A speed=N/A
Too many packets buffered for output stream 0:0.
Conversion failed!

And on my androidTv with jellyfin no films works...

Error code :
Stream mapping:
  Stream #0:0 -> #0:0 (mpeg4 (native) -> h264 (h264_vaapi))
  Stream #0:1 -> #0:1 (mp3 (mp3float) -> ac3 (native))
Press [q] to stop, [?] for help
[mpeg4 @ 0x55bb7f3fa800] No support for codec mpeg4 profile 15.
[mpeg4 @ 0x55bb7f3fa800] Failed setup for format vaapi_vld: hwaccel initialisation r
[mpeg4 @ 0x55bb7f3f6400] Video uses a non-standard and wasteful way to store B-frameder using the mpeg4_unpack_bframes bitstream filter without encoding but stream copy
[h264_vaapi @ 0x55bb7f3f8380] Driver does not support some wanted packed headers (wa
[h264_vaapi @ 0x55bb7f3f8380] Driver does not support packed sequence headers, but a
[h264_vaapi @ 0x55bb7f3f8380] No global header will be written: this may result in afor some purposes (e.g. not muxable to some containers).
Could not write header for output file #0 (incorrect codec parameters ?): Invalid daut
Error initializing output stream 0:0 --
Conversion failed!


So i'm very happy but i have a little problem with jellyfin on androidTv.

**Comment by Optimizer-Prime** [comment]
Glad that it worked for you! What AMD card do you have? I think the HEVC stuff will improve in the near future at least.

As for Android TV, I don't use it myself so don't know if I can help there. You said no films work on there, even things encoded in h264? If you disable VAAPI and go back to software encoding do you still have the same problem?

**Comment by No-Zi** [comment]
The output of vainfo with user jellyfin:

jellyfin@jelly-srv:/$ vainfo
error: XDG_RUNTIME_DIR not set in the environment.
MoTTY X11 proxy: Unsupported authorisation protocol
error: can't connect to X server!
libva info: VA-API version 1.7.0
libva info: User environment variable requested driver 'radeonsi'
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/radeonsi_drv_video.so
libva info: Found init function __vaDriverInit_1_7
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.7 (libva 2.7.0)
vainfo: Driver version: Mesa Gallium driver 20.2.0-devel for AMD Radeon (TM) RX 480 Graphics (POLARIS10, DRM 3.36.0, 5.6.0-2-amd64, LLVM 10.0.1)
vainfo: Supported profile and entrypoints
      VAProfileMPEG2Simple            : VAEntrypointVLD
      VAProfileMPEG2Main              : VAEntrypointVLD
      VAProfileVC1Simple              : VAEntrypointVLD
      VAProfileVC1Main                : VAEntrypointVLD
      VAProfileVC1Advanced            : VAEntrypointVLD
      VAProfileH264ConstrainedBaseline: VAEntrypointVLD
      VAProfileH264ConstrainedBaseline: VAEntrypointEncSlice
      VAProfileH264Main               : VAEntrypointVLD
      VAProfileH264Main               : VAEntrypointEncSlice
      VAProfileH264High               : VAEntrypointVLD
      VAProfileH264High               : VAEntrypointEncSlice
      VAProfileHEVCMain               : VAEntrypointVLD
      VAProfileHEVCMain               : VAEntrypointEncSlice
      VAProfileHEVCMain10             : VAEntrypointVLD
      VAProfileJPEGBaseline           : VAEntrypointVLD
      VAProfileNone                   : VAEntrypointVideoProc

**Comment by sum01** [comment]
Any reason why this was closed? I'm running into the same issues with the AMD RX 480.
I'm on Ubuntu 20.04 server LTS (in a QEMU VM), with `mesa-va-drivers` installed.

I tried using the `ffmpeg` from the repo's but that didn't help.
`vainfo` tells me I'm on Mesa v20.2.6 with VAAPI driver v1.7.

Doing anything other than direct play causes that error to pop up when VAAPI transcoding is enabled.
Even on 1080p H264 videos, it will happen if I set quality to, say, 480p.
